Index: gcc-4.2.4/gcc/config/sh/linux.h
===================================================================
--- gcc-4.2.4.orig/gcc/config/sh/linux.h	2008-10-09 11:01:19.000000000 +0100
+++ gcc-4.2.4/gcc/config/sh/linux.h	2008-10-09 11:02:28.000000000 +0100
@@ -63,6 +63,7 @@
    %{!static: \
      %{rdynamic:-export-dynamic} \
      %{!dynamic-linker:-dynamic-linker " LINUX_DYNAMIC_LINKER "}} \
+   --hash-style=gnu \
    %{static:-static}"
 
 /* Output assembler code to STREAM to call the profiler.  */
