diff -ruN qt-everywhere-opensource-src-4.8.0_orig/demos/browser/browserapplication.cpp qt-everywhere-opensource-src-4.8.0/demos/browser/browserapplication.cpp
--- qt-everywhere-opensource-src-4.8.0_orig/demos/browser/browserapplication.cpp	2011-11-16 14:32:14.802936599 +0100
+++ qt-everywhere-opensource-src-4.8.0/demos/browser/browserapplication.cpp	2011-11-17 10:23:19.981279222 +0100
@@ -250,6 +250,8 @@
 
     defaultSettings->setAttribute(QWebSettings::DnsPrefetchEnabled, true);
 
+    /* this is usefull for BACK/FORWARD navigation */
+    defaultSettings->setMaximumPagesInCache(5);
     settings.endGroup();
 }
 
diff -ruN qt-everywhere-opensource-src-4.8.0_orig/demos/browser/networkaccessmanager.cpp qt-everywhere-opensource-src-4.8.0/demos/browser/networkaccessmanager.cpp
--- qt-everywhere-opensource-src-4.8.0_orig/demos/browser/networkaccessmanager.cpp	2011-11-16 14:32:14.642178755 +0100
+++ qt-everywhere-opensource-src-4.8.0/demos/browser/networkaccessmanager.cpp	2011-11-17 10:23:42.255358735 +0100
@@ -90,6 +90,8 @@
     // this is a temporary hack until we properly use the pipelining flags from QtWebkit
     // pipeline everything! :)
     request.setAttribute(QNetworkRequest::HttpPipeliningAllowedAttribute, true);
+    request.setAttribute(QNetworkRequest::CacheLoadControlAttribute, QNetworkRequest::PreferNetwork);
+
     return QNetworkAccessManager::createRequest(op, request, outgoingData);
 }
 
