--- driver/e2_proc/e2_proc_main.c	2013-12-27 11:19:32.942961189 +0100
+++ driver/e2_proc/e2_proc_main_mod.c	2014-01-02 15:09:14.058362403 +0100
@@ -137,6 +137,7 @@
  *  ---------- info
  *  |           |
  *  |           --------- model <- Version String of out Box
+ *  |           --------- chipset <- Version String of chipset
  *  |
  *  ---------- tsmux
  *  |           |
@@ -300,6 +301,19 @@
   return len;
 }
 
+static int info_chipset_read(char *page, char **start, off_t off, int count,
+                           int *eof, void *data)
+{
+#if defined(SPARK)
+  int len = sprintf(page, "STi7111\n");
+#elif defined(SPARK7162)
+  int len = sprintf(page, "STi7162\n");
+#else
+  int len = sprintf(page, "unknown\n");
+#endif
+  return len;
+}
+
 static char* three_d_mode = NULL;
 
 static int three_d_mode_read(char *page, char **start, off_t off, int count,
@@ -508,6 +522,7 @@
 
 	{cProcDir  , "stb/info"                                                         , NULL, NULL, NULL, NULL, ""},
 	{cProcEntry, "stb/info/model"                                                   , NULL, info_model_read, NULL, NULL, ""},
+	{cProcEntry, "stb/info/chipset"                                                   , NULL, info_chipset_read, NULL, NULL, ""},
 
 	{cProcDir  , "stb/video"                                                        , NULL, NULL, NULL, NULL, ""},
 	{cProcEntry, "stb/video/alpha"                                                  , NULL, NULL, NULL, NULL, ""},
